babel/src/board.js

```diff
@@ -283,20 +280,29 @@ export class Board {
         let hasWideStyleMatch = _.some(_.map(flatChunks, hasMatchInSingleRow));
         return hasWideStyleMatch || _.some(_.map(chunks), hasMatchInPairOfRows);
     }
+    
+    hasMatchEvent() {
+        if (findMatches(this.orbs)[0]) { return true };
+    }
 
     swap(swapOrbs) {
-        let [[x1, y1], [x2, y2]] = swapOrbs;
+        let [[row1, col1], [row2, col2]] = swapOrbs;
         let orbsBefore = _.cloneDeep(this.orbs);
-        this.orbs[x1][y1] = orbsBefore[x2][y2]
-        this.orbs[x2][y2] = orbsBefore[x1][y1]
+        this.orbs[row1][col1] = orbsBefore[row2][col2]
+        this.orbs[row2][col2] = orbsBefore[row1][col1]
 
         // undo the swap if it did not yeild a match
-        if (!findMatches(this.orbs)[0]) {
+        if (!this.hasMatchEvent()) {
             this.orbs = orbsBefore;
         };
     }
 
-    shuffle() {
-
```

Comment:
Nah, it doesn't have to drop a match. It will more than likely drop in a possible match most times (in my experience with 8x8 boards, even though it's all random it usually happens), and if it doesn't, it doesn't hurt to shuffle.

Just remember to keep shuffling until

1. There is a potential match
2. There are no evaluate-type matches

you can't shuffle a board and give it back to the player with a three+ match already set on the board.